### PR TITLE
Delete unremoveable items when removed from hands or inventory

### PR DESCRIPTION
**Pull request by brainfood1183**
Deletes an unremoveable item when that item is dropped to the floor by unusual means such as the corpse being gibbed.
This can be disabled however in proto if some situation arises in which a person would like an unremoveable item to be dropped.

https://user-images.githubusercontent.com/113240905/233807467-b351150d-1be2-473e-80c5-361863121a21.mp4

- Fixes unremoveable (cursed) items from appearing on the floor, this includes cluwne items and cat ears.
- Cluwne backpack now removes its unremoveable state when removed from inventory(gibbing) and becomes a regular backpack crew can wear.

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

### CODE REVIEW

File: Content.Shared/Interaction/Components/UnremoveableComponent.cs [hunk 1]
```diff
@@ -1,8 +1,13 @@
+using Content.Shared.Whitelist;
 using Robust.Shared.GameStates;
 
 namespace Content.Shared.Interaction.Components
 {
     [RegisterComponent]
     [NetworkedComponent]
-    public sealed class UnremoveableComponent : Component {}
+    public sealed class UnremoveableComponent : Component 
+	{
```
Comment (metalgearsloth) [inline Content.Shared/Interaction/Components/UnremoveableComponent.cs:9]: Looks likes tabs and spaces got mixed here so please fix that.

File: Content.Shared/Interaction/Components/UnremoveableComponent.cs [hunk 2]
```diff
@@ -1,8 +1,13 @@
...
+        [DataField("deleteOnDrop")]
+        public bool DeleteOnDrop = true;
```
Comment (metalgearsloth) [inline Content.Shared/Interaction/Components/UnremoveableComponent.cs:11]: Add a comment on this and clarify it's only for hand drops not container removals please.

File: Content.Shared/Interaction/SharedInteractionSystem.cs [hunk 3]
```diff
@@ -133,6 +137,15 @@ private void OnRemoveAttempt(EntityUid uid, UnremoveableComponent item, Containe
             args.Cancel();
         }
 
+        private void OnUnequipp(EntityUid uid, UnremoveableComponent item, GotUnequippedEvent args)
+        {
```
Comment (metalgearsloth) [inline Content.Shared/Interaction/SharedInteractionSystem.cs:141]: 1 p.

**Review by metalgearsloth — CHANGES_REQUESTED** [review]

File: Content.Shared/Interaction/Components/UnremoveableComponent.cs (hunk 1 above)
Comment (brainfood1183) [reply Content.Shared/Interaction/Components/UnremoveableComponent.cs]: should be fixed.

File: Content.Shared/Interaction/SharedInteractionSystem.cs (hunk 3 above)
Comment (brainfood1183) [reply Content.Shared/Interaction/SharedInteractionSystem.cs]: changed it.

File: Content.Shared/Interaction/Components/UnremoveableComponent.cs (hunk 2 above)
Comment (brainfood1183) [reply Content.Shared/Interaction/Components/UnremoveableComponent.cs:15]: added a comment, items inside can still be removed without being deleted. If target is gibbed however with items inside their backpack and backpack has DeleteOnDrop true then items inside are lost, if backpack has DeleteOnDrop = false then the backpack loses its unremoveable component.

File: Content.Shared/Interaction/SharedInteractionSystem.cs [hunk 4]
```diff
@@ -133,6 +139,25 @@ private void OnRemoveAttempt(EntityUid uid, UnremoveableComponent item, Containe
...
+            {
+                if (item.DeleteOnDrop == true)
+                {
+                    if (_net.IsServer)
+                        QueueDel(uid);
+                }
+
+                if (item.DeleteOnDrop == false)
+                    RemComp<UnremoveableComponent>(uid);
+            }
```
Comment (ElectroJr) [inline Content.Shared/Interaction/SharedInteractionSystem.cs:158]:
```suggestion
            if (!item.DeleteOnDrop)
                RemCompDeferred<UnremoveableComponent>(uid);
            else if (_net.IsServer)
                QueueDel(uid);
```

**Review by ElectroJr — COMMENTED** [review]
Eventually this should be extended to include iunremoveable hand items, but that can be done after #15664

File: Content.Shared/Interaction/SharedInteractionSystem.cs (hunk 4 above)
Comment (brainfood1183) [reply Content.Shared/Interaction/SharedInteractionSystem.cs]: Made this change. Works great.

**Pull request merged**